glusterd2/commands/volumes/volume-option.go

```diff
@@ -66,24 +66,47 @@ type txnOpType uint8
 const (
 	txnDo txnOpType = iota
 	txnUndo
+	txnSet txnOpType = iota
+	txnReset
 )
 
 func xlatorActionDoSet(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnDo)
+	return xlatorAction(c, txnDo, txnSet)
 }
 
 func xlatorActionUndoSet(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnUndo)
+	return xlatorAction(c, txnUndo, txnSet)
+}
+
+func xlatorActionDoReSet(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnDo, txnReset)
+}
+
+func xlatorActionUndoReSet(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnUndo, txnReset)
 }
 
 // This function can be reused when volume reset operation is implemented.
 // However, volume reset can be also be treated logically as volume set but
 // with the value set to default value.
-func xlatorAction(c transaction.TxnCtx, txnOp txnOpType) error {
-
-	var req api.VolOptionReq
-	if err := c.Get("req", &req); err != nil {
-		return err
+func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, txnType txnOpType) error {
```

Comment:
The volume set or reset should be a different type altogether. One is user initiated op, the other is internal to txn framework. See earlier comment.